src/scenarios/shared/startup.py

```diff
@@ -48,7 +49,7 @@ def __init__(self):
 
     
     def _setstartuppath(self, path: str):
-        self.startupexe = os.path.join(path, 'Startup.exe')
+        self.startuppath = os.path.join(path, "Startup%s" % extension()) 
 
     def runtests(self, apptorun: str, **kwargs):
```

Comment:
hmm you mean we could do something like ```runtests(self, traits: TestTraits, scenarioname: str, scenariotypename: str, apptorun: str)```?   I'm not sure if startup should know ```TestTraits```.

Reply:
it basically already does, but it's a bunch of strings instead of a type. :)

Reply:
hmm I'm fine with having a bunch of string params for ```runtests()```, just not sure how does that solve the problem of using ```**self.traits.get_all_traits( )```? I assume we still need to do ```startup.runtests(**self.traits.get_all_traits( ), <non-trait attributes>)``` for convenience... or you mean use ```startup.runtests(guiapp=self.traits.guiapp, ...)```

Reply:
I mean that the current implementation of `runtests` looks like this (in this PR):
```
	        startup_args = [
            self.startupexe,	            self.startuppath,
            '--app-exe', apptorun,	            '--app-exe', apptorun,
            '--metric-type', kwargs['startupmetric'], 	            '--metric-type', kwargs['startupmetric'], 
            '--trace-file-name', '%s_startup.etl' % (kwargs['scenarioname'] or '%s_%s' % (kwargs['exename'],kwargs['scenariotypename'])),	            '--trace-name', '%s_startup' % (kwargs['scenarioname'] or '%s_%s' % (kwargs['exename'],kwargs['scenariotypename'])),
This conversation was marked as resolved by billwert
            '--process-will-exit', (kwargs['processwillexit'] or 'true'),	            '--process-will-exit', (kwargs['processwillexit'] or 'true'),
            '--iterations', '%s' % (kwargs['iterations'] or defaultiterations),	            '--iterations', '%s' % (kwargs['iterations'] or defaultiterations),
            '--timeout', '%s' % (kwargs['timeout'] or '50'),	            '--timeout', '%s' % (kwargs['timeout'] or '50'),
            '--warmup', '%s' % (kwargs['warmup'] or 'true'),	            '--warmup', '%s' % (kwargs['warmup'] or 'true'),
            '--gui-app', kwargs['guiapp'],	            '--gui-app', kwargs['guiapp'],
            '--working-dir', '%s' % (kwargs['workingdir'] or sys.path[0]),	            '--working-dir', '%s' % (kwargs['workingdir'] or sys.path[0]),
            '--report-json-path', reportjson,	            '--report-json-path', reportjson,
            '--trace-directory', TRACEDIR	            '--trace-directory', TRACEDIR
```

So the contract is already that `runtests` knows a bunch of strings which are properties on `TestTraits`. It knows about `TestTraits` whether we tell it the name in a param because it's already tighly coupled. So instead of passing `kwargs` we could pass a `TestTraits`, and for the few required things that aren't on `TestTratis` we have actual required parameters to `runtests`. Does that make sense?

Reply:
Sounds great! And we don't have to type string every time we access the kwargs dict.